lib/sso_users_api/manager.rb

```diff
@@ -13,18 +18,83 @@ def initialize(user)
     end
 
     def call
-      new_sso_user || update_sso_user
+      if use_auth0_for_user?
+        manage_auth0_user
+      else
+        manage_identity_server_user
+      end
     end
 
     private
 
+    # Determine if we should use Auth0 for this user
+    def use_auth0_for_user?
+      # New users (no sso_id) always go to Auth0
+      # Even if they have sso_provider='identity_server' from DB default
+      return true if !sso_id
+
+      # Existing users with sso_id: check their sso_provider
+      user.respond_to?(:sso_provider) && user.sso_provider == 'auth0'
+    end
+
+    # Auth0 user management
+    def manage_auth0_user
+      # Ensure access token is set
+      SsoUsersApi::Auth0Base.access_token = SsoUsersApi::Auth0AccessToken.call
+
+      new_auth0_user || update_auth0_user
+    end
+
+    def new_auth0_user
+      return if sso_id
+      return if auth0_user_exists?
+
+      response = SsoUsersApi::Auth0User.new(**base_attrs).create
+
+      if user.respond_to?(:sso_id) && response&.ID
+        update_attrs = { sso_id: response.ID }
+        update_attrs[:sso_provider] = 'auth0' if user.respond_to?(:sso_provider)
```

Comment:
move it to constant and name it 'bonterra_auth', use same constant in gem and all three apps.